src/validation.cpp

```diff
@@ -1588,7 +1588,8 @@ DisconnectResult CChainState::DisconnectBlock(const CBlock& block, const CBlockI
                 error("DisconnectBlock(): transaction and undo data inconsistent");
                 return DISCONNECT_FAILED;
             }
-            for (unsigned int j = tx.vin.size(); j-- > 0;) {
+            for (unsigned int j = tx.vin.size(); j > 0;) {
```

Comment:
It doesn't matter if it wraps, `j` is only used inside the loop and it only wraps when `j = 0` which is when the loop ends.

Anyway, how about changing to:
```cpp
for (int j = tx.vin.size() - 1; j >= 0; --j) {
```
which IMO is more clear.

Reply:
All unsigned integer overflows matter :-)